junit-jupiter-api/src/main/java/org/junit/jupiter/api/AfterAll.java

```diff
@@ -36,6 +36,7 @@
  * {@link TestInstance @TestInstance(Lifecycle.PER_CLASS)}. {@code @AfterAll}
  * methods may optionally declare parameters to be resolved by
  * {@link org.junit.jupiter.api.extension.ParameterResolver ParameterResolvers}.
+ * {@code @AfterAll} methods must not return a value.
```

Comment:
I would move this new requirement to the first sentence and reword it something along the lines of:

```
{@code @AfterAll} methods must have a {@code void} return type, must not be {@code private}, and must be {@code static} by default.
```

And I would do the same for the rest of the related JavaDoc updates.